sources/packages/api/src/services/restriction/institution-restriction.service.ts

```diff
@@ -0,0 +1,190 @@
+import { Injectable, Inject } from "@nestjs/common";
+import { RecordDataModelService } from "../../database/data.model.service";
+import { Connection } from "typeorm";
+import { RESTRICTION_NOT_ACTIVE } from "./student-restriction.service";
+
+import {
+  InstitutionRestriction,
+  Restriction,
+  User,
+  Note,
+  NoteType,
+  Institution,
+} from "../../database/entities";
+import {
+  AssignRestrictionDTO,
+  ResolveRestrictionDTO,
+} from "../../route-controllers/restriction/models/restriction.dto";
+import { CustomNamedError } from "../../utilities";
+
+/**
+ * Service layer for institution Restriction.
+ */
+@Injectable()
+export class InstitutionRestrictionService extends RecordDataModelService<InstitutionRestriction> {
+  constructor(@Inject("Connection") connection: Connection) {
+    super(connection.getRepository(InstitutionRestriction));
+  }
+
+  /**
+   * Service method to get all restrictions as a summary for an institution.
+   * @param institutionId
+   * @returns Institution restrictions.
+   */
+  async getInstitutionRestrictionsById(
+    institutionId: number,
+  ): Promise<InstitutionRestriction[]> {
+    return this.repo
+      .createQueryBuilder("institutionRestrictions")
+      .select([
+        "institutionRestrictions.id",
+        "institutionRestrictions.isActive",
+        "institutionRestrictions.updatedAt",
+        "institutionRestrictions.createdAt",
+        "restriction.restrictionType",
+        "restriction.restrictionCategory",
+        "restriction.description",
+      ])
+      .innerJoin("institutionRestrictions.restriction", "restriction")
+      .innerJoin("institutionRestrictions.institution", "institution")
+      .where("institution.id = :institutionId", { institutionId })
+      .orderBy("institutionRestrictions.isActive", "DESC")
+      .getMany();
+  }
+
+  /**
+   * Get institution restriction detail.
+   * @param institutionId
+   * @param restrictionId
+   * @returns
+   */
+  async getInstitutionRestrictionDetailsById(
+    institutionId: number,
+    institutionRestrictionId: number,
+  ): Promise<InstitutionRestriction> {
+    return this.repo
+      .createQueryBuilder("institutionRestrictions")
+      .select([
+        "institutionRestrictions.id",
+        "institutionRestrictions.isActive",
+        "institutionRestrictions.updatedAt",
+        "institutionRestrictions.createdAt",
+        "creator.firstName",
+        "creator.lastName",
+        "modifier.firstName",
+        "modifier.lastName",
+        "restriction.restrictionType",
+        "restriction.restrictionCategory",
+        "restriction.description",
+        "restrictionNote.description",
+        "resolutionNote.description",
+      ])
+      .innerJoin("institutionRestrictions.restriction", "restriction")
+      .leftJoin("institutionRestrictions.creator", "creator")
```

Comment:
i didn't get this part @dheepak-aot . can you explain? I believe creator and modifiers are in relation with user table

Reply:
Yes. creator is the alias that I am using for user table here. Like restriction is the alias used for restriction table.